langlib/langlib-test/src/test/resources/test-src/floatlib_test.bal

```diff
@@ -415,6 +415,733 @@ function testFromStringNegative() {
+    assertEquality("-5.64000000e-8", float:toExpString(-0.0000000564f, 8));
+    assertEquality("-5.6400000e-8", float:toExpString(-0.0000000564F, 7));
+    assertEquality("0.00000e+0", float:toExpString(-0f, 5));
+    assertEquality("0.000000000000000e+0", float:toExpString(-0F, 15));
+}
+
+function testToExpStringWithVerySmallAndLargeNegativeFloats() {
+    float a = -2.5E-17;
+    assertEquality("-2.49999999999999995e-17", float:toExpString(a, 17));
+    assertEquality("-2.5000000000e-17", float:toExpString(a, 10));
+    assertEquality("-2.5000e-17", float:toExpString(a, 4));
+
+    float b = -2.5E+17;
+    assertEquality("-2.50000000000000000e+17", float:toExpString(b, 17));
+    assertEquality("-2.5000000000e+17", float:toExpString(b, 10));
+    assertEquality("-2.5000e+17", float:toExpString(b, 4));
+
+    float c = -1.423223E6;
+    assertEquality("-1.42322300000000000e+6", c.toExpString(17));
+    assertEquality("-1.4232230000e+6", c.toExpString(10));
+    assertEquality("-1.4232e+6", c.toExpString(4));
+
+    float d = -0.9E-10f;
+    assertEquality("-8.99999999999999994e-11", d.toExpString(17));
+    assertEquality("-9.000000000000e-11", d.toExpString(12));
+    assertEquality("-9.0000e-11", d.toExpString(4));
+}
+
+function testToExpStringWithHexaDecimalFloatingPoints() {
+    float a = 0xabcff;
+    assertEquality("7.03743000000000000e+5", a.toExpString(17));
+    assertEquality("7.0374300000e+5", a.toExpString(10));
+    assertEquality("7.0374e+5", a.toExpString(4));
+
+    float b = 0xAB126fa;
+    assertEquality("1.79382010000000000e+8", b.toExpString(17));
+    assertEquality("1.7938201000e+8", b.toExpString(10));
+    assertEquality("1.7938e+8", b.toExpString(4));
+
+    a = -0xabcff;
+    assertEquality("-7.03743000000000000e+5", a.toExpString(17));
+    assertEquality("-7.0374300000e+5", a.toExpString(10));
+    assertEquality("-7.0374e+5", a.toExpString(4));
+
+    b = -0xAB126fa;
+    assertEquality("-1.79382010000000000e+8", b.toExpString(17));
+    assertEquality("-1.7938201000e+8", b.toExpString(10));
+    assertEquality("-1.7938e+8", b.toExpString(4));
+
+    a = 0x1.e412904862198p4;
+    assertEquality("3.02545321300000012e+1", a.toExpString(17));
+    assertEquality("3.0254532130e+1", a.toExpString(10));
+    assertEquality("3.0254e+1", a.toExpString(4));
+
+    b = 0x1.cbe3p17;
+    assertEquality("2.35462000000000000e+5", b.toExpString(17));
+    assertEquality("2.3546200000e+5", b.toExpString(10));
+    assertEquality("2.3546e+5", b.toExpString(4));
+
+    a = -0x1.e412904862198p4;
+    assertEquality("-3.02545321300000012e+1", a.toExpString(17));
+    assertEquality("-3.0254532130e+1", a.toExpString(10));
+    assertEquality("-3.0254e+1", a.toExpString(4));
+
+    b = -0x1.cbe3p17;
+    assertEquality("-2.35462000000000000e+5", b.toExpString(17));
+    assertEquality("-2.3546200000e+5", b.toExpString(10));
+    assertEquality("-2.3546e+5", b.toExpString(4));
+}
```

Comment:
Spec says, 
> If `x` is zero, the exponent is zero. A zero exponent is represented with a `+` sign.

Did we cover that? 
I'm getting a runtime exception for that.
```bal
    float a = 0;
    string s = a.toExpString(());
```

```
Error Output: error: java.lang.NumberFormatException {"message":"Character N is neither a decimal digit number, decimal point, nor "e" notation exponential mark."}
	at ballerina.lang.float.0:toExpString(float.bal:428)
```

Reply:
Did cover, but seems not properly handled when the fractionDigits is (). will fix.

Reply:
Fixed in d86a63e